aws_lambda_powertools/event_handler/appsync.py

```diff
@@ -28,69 +261,109 @@ def resolver(self, type_name: str = "*", field_name: Optional[str] = None):
+        self._batch_async_resolver_registry: BaseResolverRegistry = ResolverRegistry()
+        self._router_context: RouterContext = RouterContext()
 
-    **Sample usage**
+    # Interfaces
+    def resolver(self, type_name: str = "*", field_name: Optional[str] = None) -> Callable:
+        return self._resolver_registry.resolver(field_name=field_name, type_name=type_name)
 
-        from aws_lambda_powertools.event_handler import AppSyncResolver
+    def batch_resolver(self, type_name: str = "*", field_name: Optional[str] = None) -> Callable:
+        return self._batch_resolver_registry.resolver(field_name=field_name, type_name=type_name)
+
+    def batch_async_resolver(self, type_name: str = "*", field_name: Optional[str] = None) -> Callable:
+        return self._batch_async_resolver_registry.resolver(field_name=field_name, type_name=type_name)
+
+    def append_context(self, **additional_context) -> None:
+        self._router_context.context = additional_context
 
-        app = AppSyncResolver()
 
-        @app.resolver(type_name="Query", field_name="listLocations")
-        def list_locations(page: int = 0, size: int = 10) -> list:
-            # Your logic to fetch locations with arguments passed in
-            return [{"id": 100, "name": "Smooth Grooves"}]
-
-        @app.resolver(type_name="Merchant", field_name="extraInfo")
-        def get_extra_info() -> dict:
-            # Can use "app.current_event.source" to filter within the parent context
-            account_type = app.current_event.source["accountType"]
-            method = "BTC" if account_type == "NEW" else "USD"
-            return {"preferredPaymentMethod": method}
-
-        @app.resolver(field_name="commonField")
-        def common_field() -> str:
-            # Would match all fieldNames matching 'commonField'
-            return str(uuid.uuid4())
+class AppSyncResolver(Router):
     """
+    AppSync GraphQL API Resolver
 
-    def __init__(self):
+    Example
+    -------
+    ```python
+    from aws_lambda_powertools.event_handler import AppSyncResolver
+
+    app = AppSyncResolver()
+
+    @app.resolver(type_name="Query", field_name="listLocations")
+    def list_locations(page: int = 0, size: int = 10) -> list:
+        # Your logic to fetch locations with arguments passed in
+        return [{"id": 100, "name": "Smooth Grooves"}]
+
+    @app.resolver(type_name="Merchant", field_name="extraInfo")
+    def get_extra_info() -> dict:
+        # Can use "app.current_event.source" to filter within the parent context
+        account_type = app.current_event.source["accountType"]
+        method = "BTC" if account_type == "NEW" else "USD"
+        return {"preferredPaymentMethod": method}
+
+    @app.resolver(field_name="commonField")
+    def common_field() -> str:
+        # Would match all fieldNames matching 'commonField'
+        return str(uuid.uuid4())
+    ```
+    """
+
+    def __init__(self, raise_error_on_failed_batch: bool = False):
```

Comment:
```suggestion
    def __init__(self):
```

Move this to the `batch_resolver` decorator and name it `raise_on_error=True`.

**Why**

This will allow customers to have fine-grained control. For example, they might have more than one batch operations, where one could fail gracefully (partial error) while critical ones should fail upon the first failure.

This solves both. As a bonus, it shortens the flag, makes it more explicit, and it decouples from Direct Invocation ;)